File: galgamecommon/src/main/scala/com/goexp/galgame/common/model/game/GameCharacter.java
package com.goexp.galgame.common.model.game;

import java.util.StringJoiner;

public class GameCharacter {
    public String name;

    public String cv;

    public String intro;

    public String trueCV;

    public String img;
    public int index;

    @Override
    public String toString() {
        return new StringJoiner(", ", "\n" + GameCharacter.class.getSimpleName() + "[", "]")
                .add("" + index)
                .add("'" + name + "'")
                .add("cv='" + cv + "'")
//                    .add("intro='" + intro + "'")
                .add("trueCV='" + trueCV + "'")
//                    .add("img='" + img + "'")
                .toString();
    }
}

File: galgamecommon/src/main/scala/com/goexp/galgame/common/model/game/GameCharacter.scala
package com.goexp.galgame.common.model.game

import com.goexp.common.util.string.Strings

case class GameCharacter(name: String,
                         cv: String,
                         intro: String,
                         trueCV: String,
                         img: String,
                         index: Int) {


  override def toString = s"GameCharacter([$index]$name, $cv, $trueCV, $img)"

  def getShowCV(): Option[String] = {
    if (Strings.isNotEmpty(trueCV)) Some(s"*$trueCV")
    else Some(cv).filter(Strings.isNotEmpty)
  }
}